gagan-maheshwari/todo-backend
# Feature/create todo item

## Description
Wrote the function to create a new Todo item

## Database schema changes
1. Created a new database: todo-app-db
2. Added a new table: todos

## Tests
### Automated test cases added
1. should create a new todo item
2. should disallow blank todo item
3. should disallow any duplicate todo item

### Manual test cases run

*Pre-requisites*
1. Run the application using `npm run build` followed by `npm run start`
2. Open Postman and create new request with the following details:
> 1. Method Type: POST
> 2. URL: http://localhost:4200/todo/todos
> 3. In the Body section, select `raw` and `JSON`

**Test Case 1**
Verify the happy case.

1. Enter the data as:
`
{
 "title": "New Todo"
}
`
2. Hit `Send` button
3. Verify ER1.

Expected Result:
1. The output have the following:
> Status: 201 Created
> A JSON containing the id and the title should come up
---
**Test Case 2**
Verify the duplicate input.

1. Execute test case 1.
2. Enter the data as:
`
{
 "title": "New Todo"
}
`
3. Hit `Send` button
4. Verify ER1.

Expected Result:
1. The output have the following:
> Status: 400 Bad Request
> A JSON containing the failure message should come up as: `This todo item already exists. Please try a new one.`
---
**Test Case 3**
Verify for blank input.

1. Enter the data as:
`
{
 "title": ""
}
`
2. Hit `Send` button
3. Verify ER1.

Expected Result:
1. The output have the following:
> Status: 400 Bad Request
> A JSON containing the failure message should come up as: `Please enter a valid todo item`
---
